packages/action-menu/src/ActionMenuApi.ts

```diff
@@ -36,7 +36,7 @@ export class ActionMenuApi {
   private agentContext: AgentContext
 
   public constructor(
-    dispatcher: Dispatcher,
+    messageHandlerRegistry: MessageHandlerRegistry,
```

Comment:
Do you think we should inject the  `DependencyManager` and register the handlers on that class?

Reply:
Right, for extension modules `DependencyManager` is more convenient. In such case I think there is no need to inject it, as it can be retrieved from `AgentContext`.

Actually I think it can be used in all the case. Or do you think that conceptually would be better to only use it out of the core?

Reply:
> Actually I think it can be used in all the case. Or do you think that conceptually would be better to only use it out of the core?

I think if we see the `DependencyManager` as the public api for extension module is not a bad idea. That can wrap the message handler registry and we'll be free to make changes to that class, but changes to the the `DependencyManager` will be seen as breaking